# Capstone 2: Business Analysis with Python

## EmporiUm Sales Analysis

### This project will show analysis for the South region with the teritorries of Florida and Texas using python and python functions like Pandas, Matplotlib and NumPy
* ##### The manager for Florida is Lana Ilana with stores 719-729
* ##### The manager for Texas is Jeff "Howdy" Richards with store 901-911

The analysis will show: 
* Monthly sales trends over time
* Ranking of store performance
* Analysis of customer buying behavior
* Evaluation of product category performance
* Data-driven marketing strategies

In [139]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import datetime

In [140]:
StoreDetail = pd.read_csv('StoreDetail.csv')

In [141]:
StoreDetail.head()

,Store Location,State,Store ID,Territory Manager,Region,Region Director
0,Aurora,Colorado,701,Jim Heck,West,Cassie Chambers
1,Berthoud,Colorado,702,Jim Heck,West,Cassie Chambers
2,Boulder,Colorado,703,Jim Heck,West,Cassie Chambers
3,Castle Rock,Colorado,704,Jim Heck,West,Cassie Chambers
4,Denver,Colorado,705,Jim Heck,West,Cassie Chambers


In [142]:
StoreDetail.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 111 entries, 0 to 110
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Store Location     111 non-null    object
 1   State              111 non-null    object
 2   Store ID           111 non-null    int64 
 3   Territory Manager  111 non-null    object
 4   Region             111 non-null    object
 5   Region Director    111 non-null    object
dtypes: int64(1), object(5)
memory usage: 5.3+ KB


In [143]:
StoreSales = pd.read_csv('StoreSales.csv')

In [144]:
StoreSales.head()

,Transaction Date,Store ID,RewardsID,Prod Num,Sale Amount
0,1/1/2022,702,NaN,105349-M,8.00
1,1/1/2022,704,NaN,105350-T,144.00
2,1/1/2022,705,NaN,105351-M,44.00
3,1/1/2022,705,NaN,105352-M,47.61
4,1/1/2022,705,NaN,105353-A,20.36


In [145]:
StoreSales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 335129 entries, 0 to 335128
Data columns (total 5 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Transaction Date  335129 non-null  object 
 1   Store ID          335129 non-null  int64  
 2   RewardsID         34943 non-null   float64
 3   Prod Num          335129 non-null  object 
 4   Sale Amount       335129 non-null  float64
dtypes: float64(2), int64(1), object(2)
memory usage: 12.8+ MB


In [146]:
Products = pd.read_csv('Products.csv')

In [147]:
Products.head()

,Prod Num,Product,CategoryID,SubcategoryID
0,105248-IT,TCL NXTPAPER 10s,120,120-tab
1,105249-IT,Dell Latitude 7320 Detachable,120,120-tab
2,105250-IT,Realme Pad,120,120-tab
3,105251-IT,Lenovo Tab P12 Pro,120,120-tab
4,105252-IT,Microsoft Surface Pro 9,120,120-tab


In [148]:
Products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 669 entries, 0 to 668
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Prod Num       669 non-null    object
 1   Product        669 non-null    object
 2   CategoryID     669 non-null    int64 
 3   SubcategoryID  669 non-null    object
dtypes: int64(1), object(3)
memory usage: 21.0+ KB


In [149]:
ProductCat = pd.read_csv('ProductCategories.csv')

In [150]:
ProductCat.head()

,CategoryID,Category,SubcategoryID,Subcategory
0,120,Technology & Accessories,120-tab,Tablets
1,120,Technology & Accessories,120-cal,Calculators
2,120,Technology & Accessories,120-sof,Software Download
3,120,Technology & Accessories,120-hea,Headphones
4,120,Technology & Accessories,120-ext,External Accessories


In [151]:
ProductCat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52 entries, 0 to 51
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   CategoryID     52 non-null     int64 
 1   Category       52 non-null     object
 2   SubcategoryID  52 non-null     object
 3   Subcategory    52 non-null     object
dtypes: int64(1), object(3)
memory usage: 1.8+ KB


In [152]:
CustList = pd.read_csv('customer_list.csv')

In [153]:
CustList.head()

,cust_id|date|time|name|email|phone|sms-opt-out
0,1|2023-03-15|08:45:12|Rachel|rachel@centralper...
1,2|2023-05-22|12:30:45|R. Geller|rossg@centralp...
2,3|2023-07-09|18:15:27|Monica Geller|chefmonica...
3,4|2023-09-01|21:05:33|Chandler Bing|chandlerb@...
4,5|2023-11-18|14:22:10|Joey|howyoudoing@central...


In [154]:
CustList.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 521 entries, 0 to 520
Data columns (total 1 columns):
 #   Column                                           Non-Null Count  Dtype 
---  ------                                           --------------  ----- 
 0   cust_id|date|time|name|email|phone|sms-opt-out   521 non-null    object
dtypes: object(1)
memory usage: 4.2+ KB


#  Core Marketing Analysis

### Identify territory managers, store IDs, and store locations for each assigned territory.

In [155]:
Texas = StoreDetail[(StoreDetail['State'] == 'Texas')] # only showing Texas stores
Texas

# The manager for Florida is Lana Ilana with stores 719-729
# The manager for Texas is Jeff "Howdy" Richards with store 901-911

,Store Location,State,Store ID,Territory Manager,Region,Region Director
100,Arlington,Texas,901,"Jeff ""Howdy"" Richards",South,Andy Gisselquist
101,Austin,Texas,902,"Jeff ""Howdy"" Richards",South,Andy Gisselquist
102,Bacliff,Texas,903,"Jeff ""Howdy"" Richards",South,Andy Gisselquist
103,Baytown,Texas,904,"Jeff ""Howdy"" Richards",South,Andy Gisselquist
104,Beaumont,Texas,905,"Jeff ""Howdy"" Richards",South,Andy Gisselquist
105,Cedar Park,Texas,906,"Jeff ""Howdy"" Richards",South,Andy Gisselquist
106,Dallas,Texas,907,"Jeff ""Howdy"" Richards",South,Andy Gisselquist
107,Denton,Texas,908,"Jeff ""Howdy"" Richards",South,Andy Gisselquist
108,Desoto,Texas,909,"Jeff ""Howdy"" Richards",South,Andy Gisselquist
109,Fort Worth,Texas,910,"Jeff ""Howdy"" Richards",South,Andy Gisselquist


In [156]:
Florida = StoreDetail[(StoreDetail['State'] == 'Florida')] # only showing Florida stores
Florida

,Store Location,State,Store ID,Territory Manager,Region,Region Director
26,Cape Canaveral,Florida,719,Lana Ilana,South,Andy Gisselquist
27,Fort Lauderdale,Florida,720,Lana Ilana,South,Andy Gisselquist
28,Jacksonville,Florida,721,Lana Ilana,South,Andy Gisselquist
29,Key West,Florida,722,Lana Ilana,South,Andy Gisselquist
30,Lakeland,Florida,723,Lana Ilana,South,Andy Gisselquist
31,Miami,Florida,724,Lana Ilana,South,Andy Gisselquist
32,Naples,Florida,725,Lana Ilana,South,Andy Gisselquist
33,Orlando,Florida,726,Lana Ilana,South,Andy Gisselquist
34,Sebring,Florida,727,Lana Ilana,South,Andy Gisselquist
35,Tallahassee,Florida,728,Lana Ilana,South,Andy Gisselquist


### Monthly total revenue for in-store sales in each of the two sales territories, over the full period covered by the data

#### Monthly Sales Data for Texas Territory

In [157]:
StoreSales['Transaction Date'] = pd.to_datetime(StoreSales['Transaction Date']) # Makes transaction date a datetime data type
StoreSales['Year Month'] = StoreSales['Transaction Date'].dt.to_period('M') # month by month organization
TexasStores = StoreSales[(StoreSales['Store ID'] >= 901) & (StoreSales['Store ID'] <= 911)] # Only the Texas store IDs
TexasMonthly = (TexasStores.groupby('Year Month', as_index=False)['Sale Amount'].sum()) # Grouping by YearMonth and adding sale amount

TexasMonthly

,Year Month,Sale Amount
0,2022-01,40857.20
1,2022-02,52729.22
2,2022-03,49214.95
3,2022-04,55229.25
4,2022-05,49696.98
5,2022-06,45208.26
6,2022-07,48308.66
7,2022-08,49641.33
8,2022-09,47977.04
9,2022-10,50543.10


#### Monthly Sales Data for the Florida Territory

In [158]:
StoreSales['Transaction Date'] = pd.to_datetime(StoreSales['Transaction Date']) # Makes transaction date a datetime data type
StoreSales['Year Month'] = StoreSales['Transaction Date'].dt.to_period('M') # month by month organization
FloridaStores= StoreSales[(StoreSales['Store ID'] >= 719) & (StoreSales['Store ID'] <=729)] # only Florida stores
FloridaMonthly = (FloridaStores.groupby('Year Month', as_index=False)['Sale Amount'].sum()) # adding monthly sale amount, grouping the months togther

FloridaMonthly

,Year Month,Sale Amount
0,2022-01,48143.95
1,2022-02,46257.35
2,2022-03,53541.25
3,2022-04,49505.25
4,2022-05,58326.64
5,2022-06,48683.24
6,2022-07,57545.77
7,2022-08,48656.98
8,2022-09,42839.10
9,2022-10,49508.42


### Ranking the sales performance of each store in each sales territory

#### Florida Stores ranked by top proforming
#### The top proforming store are 
* 724(Miami)
* 728(Tallahassee)
* 722(Key West)

In [159]:
FloridaStores= StoreSales[(StoreSales['Store ID'] >= 719) & (StoreSales['Store ID'] <=729)] # oblyinh using FLorida Stores
FloridaPerf = (FloridaStores.groupby('Store ID', as_index=False)['Sale Amount'].sum()) # total reveue by store
FloridaPerf = FloridaPerf.sort_values(['Sale Amount'], ascending=False) # top preofoming to least proforming
FloridaPerf = FloridaPerf.merge(Florida[['Store ID', 'Store Location']], on = 'Store ID')
FloridaPerf

,Store ID,Sale Amount,Store Location
0,724,618846.25,Miami
1,728,541670.66,Tallahassee
2,722,332620.43,Key West
3,719,331956.13,Cape Canaveral
4,721,331667.08,Jacksonville
5,727,330848.14,Sebring
6,729,308979.01,Tampa
7,720,302270.07,Fort Lauderdale
8,726,297891.71,Orlando
9,723,273835.24,Lakeland


#### Texas Stores ranked by top proforming
#### The top proforming store are 
* 905(Beaumont)
* 910(Fort Worth)
* 909(Desoto)

In [160]:
TexasStores = StoreSales[(StoreSales['Store ID'] >= 901) & (StoreSales['Store ID'] <= 911)]
TexasPerf = (TexasStores.groupby('Store ID', as_index=False)['Sale Amount'].sum()) # total reveue by store
TexasPerf = TexasPerf.sort_values(['Sale Amount'], ascending=False) # top preofoming to least proforming
TexasPerf =  TexasPerf.merge(Texas[['Store ID', 'Store Location']], on = 'Store ID') # show location name with store id 
TexasPerf

,Store ID,Sale Amount,Store Location
0,905,637818.90,Beaumont
1,910,351772.59,Fort Worth
2,909,330720.12,Desoto
3,906,312628.27,Cedar Park
4,904,298163.16,Baytown
5,908,298111.45,Denton
6,901,292987.49,Arlington
7,902,291145.00,Austin
8,903,280714.34,Bacliff
9,911,269344.64,Georgetown


### Top customers in each sales territory

#### Florida's top customers
* 47	Stanley H.	 \$4954.99
* 259	Todd Chavez	 \$4549.16
* 190	Huck	\$4415.78

In [161]:
customer_list = pd.read_csv('customer_list.csv', sep= '|') # removing | to clean up column names
customer_list

,cust_id,date,time,name,email,phone,sms-opt-out
0,1,2023-03-15,08:45:12,Rachel,rachel@centralperk.coffee,212-555-1001,N
1,2,2023-05-22,12:30:45,R. Geller,rossg@centralperk.coffee,212-555-1002,N
2,3,2023-07-09,18:15:27,Monica Geller,chefmonica@centralperk.coffee,212-555-1003,N
3,4,2023-09-01,21:05:33,Chandler Bing,chandlerb@centralperk.coffee,212-555-1004,Y
4,5,2023-11-18,14:22:10,Joey,howyoudoing@centralperk.coffee,212-555-1005,N
...,...,...,...,...,...,...,...
516,517,2024-02-15,11:30:30,Perry the Platypus,agentp@danvillekids.com,555-555-3206,N
517,518,2024-05-01,15:55:45,Baljeet,baljeet@danvillekids.com,555-555-3207,N
518,519,2023-03-05,09:20:00,Buford,buford@danvillekids.com,555-555-3208,N
519,520,2023-05-18,13:45:15,Vanessa D.,vanessa@danvillekids.com,555-555-3209,N


In [162]:
FloridaStores = FloridaStores.dropna() # removing all null rows 
FloridaStores

,Transaction Date,Store ID,RewardsID,Prod Num,Sale Amount,Year Month
29,2022-01-01,726,17.0,105381-M,55.00,2022-01
174,2022-01-02,725,61.0,105370-S,8.45,2022-01
177,2022-01-02,728,394.0,105459-IT,393.21,2022-01
309,2022-01-03,721,62.0,105552-S,7.85,2022-01
314,2022-01-03,725,439.0,105512-B,28.92,2022-01
...,...,...,...,...,...,...
334559,2025-12-30,719,223.0,105466-S,9.95,2025-12
334839,2025-12-31,719,414.0,105483-B,12.12,2025-12
334843,2025-12-31,721,405.0,105372-S,7.85,2025-12
334845,2025-12-31,721,85.0,105825-S,9.09,2025-12


In [163]:
FLCustList = FloridaStores.merge(customer_list, left_on = 'RewardsID', right_on = 'cust_id') # merging the two data frams to have name sale amount 
FLCustList.head()                                                                            # and rewards id

,Transaction Date,Store ID,RewardsID,Prod Num,Sale Amount,Year Month,cust_id,date,time,name,email,phone,sms-opt-out
0,2022-01-01,726,17.0,105381-M,55.00,2022-01,17,2024-04-22,19:25:50,Mike E.,mike@lospolloshermanos.biz,505-555-2007,N
1,2022-01-02,725,61.0,105370-S,8.45,2022-01,61,2023-02-10,09:20:15,Jake P.,jake@nypd99.com,718-555-8001,N
2,2022-01-02,728,394.0,105459-IT,393.21,2022-01,394,2023-06-13,17:00:10,Blanche Devereaux,blanche@goldengirls.house,305-555-2003,N
3,2022-01-03,721,62.0,105552-S,7.85,2022-01,62,2023-04-25,13:45:30,Amy S.,amy@nypd99.com,718-555-8002,N
4,2022-01-03,725,439.0,105512-B,28.92,2022-01,439,2024-06-06,19:25:15,David Puddy,puddy@vandelayindustries.com,212-555-2408,N


In [164]:
FLCustList = FLCustList[['RewardsID','name','Sale Amount']] # only showing the columns i need
FLCustList

,RewardsID,name,Sale Amount
0,17.0,Mike E.,55.00
1,61.0,Jake P.,8.45
2,394.0,Blanche Devereaux,393.21
3,62.0,Amy S.,7.85
4,439.0,David Puddy,28.92
...,...,...,...
3027,223.0,Callie Torres,9.95
3028,414.0,Cookie Monster,12.12
3029,405.0,Shiv Roy,7.85
3030,85.0,Dar Adal,9.09


In [165]:
FLCustList = FLCustList.groupby(['RewardsID','name'])[['Sale Amount']].sum().sort_values('Sale Amount', ascending = False)
FLCustList             # showing me the customer whos spent the most money in Florida

,,Sale Amount
RewardsID,name,
47.0,Stanley H.,4954.99
259.0,Todd Chavez,4549.16
190.0,Huck,4415.78
335.0,John Locke,3631.94
164.0,Dr. Ray Flemming,3535.88
...,...,...
234.0,Harry Morgan,19.87
268.0,Lindsay F.,12.30
470.0,Bubbles,9.45


#### Texas' top customers
* 344	Nate Jacobs	\$5240.86
* 474	Christopher Moltisanti	\$3479.46
* 102	Jodie L.	\$3464.91

In [166]:
TexasStores.dropna() # dropping null values in texas stores

,Transaction Date,Store ID,RewardsID,Prod Num,Sale Amount,Year Month
143,2022-01-01,908,328.0,105337-IT,1301.88,2022-01
279,2022-01-02,901,29.0,105487-S,9.95,2022-01
282,2022-01-02,904,444.0,105531-A,57.70,2022-01
601,2022-01-04,903,77.0,105518-M,20.36,2022-01
605,2022-01-04,905,461.0,105454-M,14.00,2022-01
...,...,...,...,...,...,...
335107,2025-12-31,905,11.0,105747-M,43.30,2025-12
335109,2025-12-31,905,231.0,105623-A,50.96,2025-12
335113,2025-12-31,906,92.0,105615-B,28.92,2025-12
335120,2025-12-31,908,114.0,105343-IT,1257.12,2025-12


In [167]:
TXCustList = TexasStores.merge(customer_list, left_on = 'RewardsID', right_on = 'cust_id') # merging the two data frames to have name sale amount 
TXCustList.head()                                                                          # and rewarda id 

,Transaction Date,Store ID,RewardsID,Prod Num,Sale Amount,Year Month,cust_id,date,time,name,email,phone,sms-opt-out
0,2022-01-01,908,328.0,105337-IT,1301.88,2022-01,328,2023-08-23,14:55:15,Monica Reyes,reyes@fbi.gov,202-555-1307,N
1,2022-01-02,901,29.0,105487-S,9.95,2022-01,29,2023-05-17,12:35:30,Samwell T.,maester@westeros.kingdom,123-555-3009,N
2,2022-01-02,904,444.0,105531-A,57.70,2022-01,444,2023-07-06,17:15:05,Charlotte York,charlotte@manhattan.nyc,212-555-2503,N
3,2022-01-04,903,77.0,105518-M,20.36,2022-01,77,2023-02-28,13:55:30,Nicholas Brody,brody@cia.gov,202-555-9102,N
4,2022-01-04,905,461.0,105454-M,14.00,2022-01,461,2023-04-23,14:15:10,Lydia Rodarte-Quayle,lydia@lospolloshermanos.biz,505-555-2610,N


In [168]:
TXCustList = TXCustList[['RewardsID','name','Sale Amount']] # only showing columns i want
TXCustList

,RewardsID,name,Sale Amount
0,328.0,Monica Reyes,1301.88
1,29.0,Samwell T.,9.95
2,444.0,Charlotte York,57.70
3,77.0,Nicholas Brody,20.36
4,461.0,Lydia Rodarte-Quayle,14.00
...,...,...,...
2488,11.0,Walter White,43.30
2489,231.0,Maria LaGuerta,50.96
2490,92.0,Elliot,28.92
2491,114.0,Regular Sized Rudy,1257.12


In [169]:
TXCustList = TXCustList.groupby(['RewardsID','name'])[['Sale Amount']].sum().sort_values('Sale Amount', ascending = False)
TXCustList                      # showing me who spent the most money in the texas stores 

,,Sale Amount
RewardsID,name,
344.0,Nate Jacobs,5240.86
474.0,Christopher Moltisanti,3479.46
102.0,Jodie L.,3464.91
168.0,Darius,3400.35
271.0,Maeby F.,3127.53
...,...,...
324.0,Walter Skinner,10.25
393.0,Rose Nylund,9.95
148.0,Potsie W.,9.09
